Provis.Core/Services/TaskService.cs

```diff
@@ -183,5 +183,57 @@ public async Task<List<WorkerRoleDTO>> GetWorkerRoles()
 
             return result.Select(x => _mapper.Map<WorkerRoleDTO>(x)).ToList();
         }
+
+        public async Task JoinTaskAsync(TaskAssignDTO taskAssign, string userId)
+        {
+            var task = await _taskRepository.
+                Query().
+                Include(x=>x.UserTasks).
+                SingleOrDefaultAsync(p => p.Id == taskAssign.Id);
+
+            if (task.TaskCreatorId == userId) // If creator of task want to assign somebody
+            {
+                List<UserTask> userTasks = new();
+                foreach (var item in taskAssign.AssignedUsers)
+                {
+                    if (userTasks.Exists(x => x.UserId == item.UserId))
+                    {
+                        throw new HttpException(System.Net.HttpStatusCode.Forbidden,
+                            "This user has already assigned");
+                    }
+                    if (task.UserTasks.Exists(x=>x.UserId == item.UserId))
+                    {
+                        throw new HttpException(System.Net.HttpStatusCode.Forbidden,
+                            "This user alredy in this task");
+                    }
+                    userTasks.Add(new UserTask
+                    {
+                        TaskId = task.Id,
+                        UserId = item.UserId,
+                        UserRoleTagId = item.RoleTagId
+                    });
+                }
+                await _userTaskRepository.AddRangeAsync(userTasks);
+                await _userTaskRepository.SaveChangesAsync();
+            }
+            else // If somebody want to assign himself on the task
+            {
+                if (task.UserTasks.Exists(x => x.UserId == userId))
+                {
+                    throw new HttpException(System.Net.HttpStatusCode.Forbidden,
+                            "You are alredy in this task");
+                }
+
+                UserTask userTask = new()
+                {
+                    TaskId = task.Id,
+                    UserId = userId,
+                    UserRoleTagId = taskAssign.AssignedUsers.SingleOrDefault().RoleTagId
```

Comment:
Why we take single or default user ?

Reply:
Because this part of method runed only where user joined himself on task. Thats mean that only one user can be in taskAssign.AssignedUsers